internal/pkg/agent/cmd/container.go

```diff
@@ -787,6 +787,13 @@ func setPaths(statePath, configPath, logsPath string, writePaths bool) error {
 	if err := syncDir(paths.Downloads(), destDownloads); err != nil {
 		return fmt.Errorf("syncing download directory to STATE_PATH(%s) failed: %w", statePath, err)
 	}
+
+	// sync components to data directory
+	destComponents := filepath.Join(statePath, "data", "components")
+	if err := syncDir(paths.Components(), destComponents); err != nil {
```

Comment:
Rather than copying all the binaries, would it make more sense to setup a symlink to do this?

Reply:
setting up a symlink would from host to container would die on container restart.

Reply:
Ah yes I forgot that the STATE_PATH could be a mounted host volume. Can you add a comment here explaining 1) why we need to do this copy 2) why we can't use a symlink and have to copy the files over.

I am a little bit concerned that the agent as a container possibly needs to copy over a significant number of files each time it restarts. Is there a way for us to avoid doing this copying unnecessarily?

Reply:
Originally we copied the downloads folder to the state path, because Elastic Agent could download binaries. With V2 we will never download a binary, even for upgrade (as containers cannot self-upgrade).

We should just alter the reading path in Elastic Agent so when its running in container mode it is in the container. Copying the binaries to this path will also run into an issue in Kubernetes. Because in Kubernetes state paths are normally mounted with `noexec`, so the Elastic Agent will not be able to spawn the binaries once it is copied over.

Reply:
Yah, seconding the sentiment here. Logic seems good based on my understanding of the issue, but we might run into issues with just copying files.